examples/interaction_fingerprint.py

```diff
@@ -0,0 +1,284 @@
+import deepchem as dc
+import json
+import numpy as np
+
+from copy import deepcopy
+from functools import partial
+from hyperopt import hp, fmin, tpe
+from shutil import copyfile
+from sklearn.ensemble import RandomForestRegressor
+
+from utils import init_trial_path, load_dataset, EarlyStopper
+
+
+def rf_model_builder(model_dir, hyperparams, mode):
+  if mode == 'regression':
+    sklearn_model = RandomForestRegressor(
+        n_estimators=hyperparams['n_estimators'],
+        criterion=hyperparams['criterion'],
+        min_samples_split=hyperparams['min_samples_split'],
+        bootstrap=hyperparams['bootstrap'])
+  return dc.models.SklearnModel(sklearn_model, model_dir)
+
+
+def load_model(save_path, n_features, args, tasks, hyperparams):
+  if args['dataset'] in ['PDBbind']:
+    mode = 'regression'
+  else:
+    raise ValueError('Unexpected dataset: {}'.format(args['dataset']))
+
+  if args['model'] == 'RF':
+    # model = dc.models.SingletaskToMultitask(
+    #     tasks, partial(rf_model_builder, hyperparams=hyperparams, mode=mode))
+    sklearn_model = RandomForestRegressor(
+        n_estimators=hyperparams['n_estimators'],
+        criterion=hyperparams['criterion'],
+        min_samples_split=hyperparams['min_samples_split'],
+        bootstrap=hyperparams['bootstrap'])
+
+    model = dc.models.SklearnModel(sklearn_model)
+  elif args['model'] == 'MTR':
+    model = dc.models.MultitaskRegressor(
+        n_tasks=1,
+        n_features=n_features,
+        layer_sizes=hyperparams['layer_sizes'],
+        dropouts=hyperparams['dropout'],
+        learning_rate=hyperparams['lr'],
+        model_dir=save_path)
+  else:
+    raise ValueError('Unexpected model: {}'.format(args['model']))
+
+  return model
+
+
+def main(save_path, args, hyperparams):
+  # Dataset
+  args, tasks, all_dataset, transformers = load_dataset(args)
+  train_set, val_set, test_set = all_dataset
+  n_features = train_set.X.shape[1]
+
+  # Metric
+  if args['metric'] == 'roc_auc':
+    metric = dc.metrics.Metric(dc.metrics.roc_auc_score, np.mean)
+  elif args['metric'] == 'rmse':
+    metric = dc.metrics.Metric(dc.metrics.rms_score, np.mean)
+  elif args['metric'] == 'r2':
+    metric = dc.metrics.Metric(dc.metrics.pearson_r2_score, np.mean)
+  else:
+    raise ValueError('Unexpected metric: {}'.format(args['metric']))
+
+  all_run_val_metrics = []
+  all_run_test_metrics = []
+
+  for _ in range(args['num_runs']):
+    # Model
+    model = load_model(save_path, n_features, args, tasks, hyperparams)
+
+    if args['model'] == 'RF':
+      model.fit(train_set)
+    else:
+      # Object for early stop tracking
+      stopper = EarlyStopper(save_path, args['metric'], args['patience'])
+
+      # 50 for maximum number of epochs
+      for epoch in range(50):
```

Comment:
Why 50 rather than 100?

Reply:
Same as above - these models overfit to pdbbind very quickly and in practice I've never found that > 20 epochs are needed. Also the training is pretty time consuming when doing hyperparam optimization.